### ISSUE TITLE
Regression in 1.1.338 with unnecessary isinstance check

### ISSUE THREAD

**Issue by kyle-query** [microsoft/pyright#6656]
**Describe the bug**

Pyright is incorrectly reporting that an `isinstance` check is unnecessary. This bug occurs with 1.1.338 but not 1.1.337.

**Code or Screenshots**

```python
import typing
from typing import Sequence, Collection

def foo0(t: type):
    if typing.get_origin(t) not in {typing.Sequence, typing.Collection, tuple, list}:
        print("...")


def foo1(t: type):
    if typing.get_origin(t) not in {Sequence, Collection, tuple, list}:
        print("...")
```

```
$ pyright --version
1.1.338

$ pyright example.py
example.py
  example.py:5:37 - error: Type of "Sequence" is partially unknown
    Type of "Sequence" is "type[Sequence[Unknown]]" (reportUnknownMemberType)
  example.py:5:54 - error: Type of "Collection" is partially unknown
    Type of "Collection" is "type[Collection[Unknown]]" (reportUnknownMemberType)
  example.py:5:8 - error: Expression will always evaluate to True since the types "ParamSpec" and "type[Sequence[Unknown]] | type[Collection[Unknown]] | type[tuple[Unknown, ...]] | type[list[Unknown]]" have no overlap (reportUnnecessaryContains)
  example.py:10:8 - error: Expression will always evaluate to True since the types "ParamSpec" and "type[Sequence[Unknown]] | type[Collection[Unknown]] | type[tuple[Unknown, ...]] | type[list[Unknown]]" have no overlap (reportUnnecessaryContains)
```

I think the error about partially unknown types on `typing.Sequence` but not `Sequence` has been reported in [another issue](https://github.com/microsoft/pyright/issues/6582).

**Comment by erictraut** [comment]
Thanks for reporting. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.339, which I just published. It will also be included in a future release of pylance.
